tests/unit/utils/test_component_tools.py

```diff
@@ -38,6 +38,19 @@ def tearDownClass(cls):
         """suppress temp dir"""
         shutil.rmtree(cls.component_matcher.gitrepo.checkoutdir)
 
+    def test_get_meta_for_file(self):
```

Comment:
It seems this unit test will clone ansible/ansible repository twice:
- one shallow clone (thanks to `GitShallowRepo`) for `AnsibleComponentMatcher`
- one full clone for `FileIndexer`

I suggest:
1. Update `GitShallowRepo.update_checkout` -> always `return False` (instead of raising an exception)
2. headers:
```
import six
six.add_move(six.MovedModule('mock', 'mock', 'unittest.mock'))
from six.moves import mock
```
3. update `setUpClass` method
```python
@classmethod
def setUpClass(cls):
    """Init the matcher"""
    cachedir = tempfile.mkdtemp()
    gitrepo = GitShallowRepo(cachedir=cachedir, repo=ComponentMatcher.REPO)
    gitrepo.update()

    @mock.patch.object(FileIndexer, 'manage_checkout')
    @mock.patch.object(FileIndexer, 'checkoutdir', create=True, side_effect=gitrepo.checkoutdir)
    def get_file_indexer(m_manage_checkout, m_checkoutdir):
        indexer = FileIndexer()
        indexer.get_files()
        indexer.parse_metadata()
        return indexer

    cls.component_matcher = ComponentMatcher(email_cache={}, gitrepo=gitrepo, file_indexer=get_file_indexer())
```
4. in `tearDownClass`: `shutil.rmtree(cls.component_matcher.gitrepo.checkoutdir)`

Reply:
Good catch, thanks!